### PR TITLE
Delete URLUtilsReadOnly, move members to WorkerLocation

### PR DESCRIPTION
**Pull request by sideshowbarker**
https://github.com/whatwg/html/commit/32a7a20 completely removed the `URLUtilsReadOnly` interface and moved/inlined/hoisted its members up into the `WorkerLocation` interface. So this change deletes `URLUtilsReadOnly`, and moves its members to `WorkerLocation`.

This change also adds spec URLs to the `WorkerLocation` data.

Related MDN change: https://github.com/mdn/content/pull/4772

### CODE REVIEW

File: api/WorkerLocation.json [hunk 1]
```diff
@@ -46,6 +47,496 @@
           "standard_track": true,
           "deprecated": false
         }
+      },
+      "hash": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/URLUtilsReadOnly/hash",
```
Comment (Elchi3) [inline api/WorkerLocation.json:53]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/WorkerLocation/hash",
```

File: api/WorkerLocation.json [hunk 2]
```diff
@@ -46,6 +47,496 @@
...
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "host": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/URLUtilsReadOnly/host",
```
Comment (Elchi3) [inline api/WorkerLocation.json:102]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/WorkerLocation/host",
```

File: api/WorkerLocation.json [hunk 3]
```diff
@@ -46,6 +47,496 @@
...
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "hostname": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/URLUtilsReadOnly/hostname",
```
Comment (Elchi3) [inline api/WorkerLocation.json:151]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/WorkerLocation/hostname",
```

File: api/WorkerLocation.json [hunk 4]
```diff
@@ -46,6 +47,496 @@
...
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "href": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/URLUtilsReadOnly/href",
```
Comment (Elchi3) [inline api/WorkerLocation.json:200]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/WorkerLocation/href",
```

File: api/WorkerLocation.json [hunk 5]
```diff
@@ -46,6 +47,496 @@
...
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "origin": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/URLUtilsReadOnly/origin",
```
Comment (Elchi3) [inline api/WorkerLocation.json:249]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/WorkerLocation/origin",
```

File: api/WorkerLocation.json [hunk 6]
```diff
@@ -46,6 +47,496 @@
...
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "pathname": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/URLUtilsReadOnly/pathname",
```
Comment (Elchi3) [inline api/WorkerLocation.json:298]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/WorkerLocation/pathname",
```

File: api/WorkerLocation.json [hunk 7]
```diff
@@ -46,6 +47,496 @@
...
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "port": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/URLUtilsReadOnly/port",
```
Comment (Elchi3) [inline api/WorkerLocation.json:347]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/WorkerLocation/port",
```

File: api/WorkerLocation.json [hunk 8]
```diff
@@ -46,6 +47,496 @@
...
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "protocol": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/URLUtilsReadOnly/protocol",
```
Comment (Elchi3) [inline api/WorkerLocation.json:396]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/WorkerLocation/protocol",
```

File: api/WorkerLocation.json [hunk 9]
```diff
@@ -46,6 +47,496 @@
...
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "search": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/URLUtilsReadOnly/search",
```
Comment (Elchi3) [inline api/WorkerLocation.json:445]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/WorkerLocation/search",
```

File: api/WorkerLocation.json [hunk 10]
```diff
@@ -46,6 +47,496 @@
...
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "toString": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/URLUtilsReadOnly/toString",
```
Comment (Elchi3) [inline api/WorkerLocation.json:494]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/WorkerLocation/toString",
```

**Review by Elchi3 — CHANGES_REQUESTED** [review]

**Review by Elchi3 — APPROVED** [review]
Thanks for helping me to get a mixin done 🎉 👍

**Comment by sideshowbarker** [comment]
> Thanks for helping me to get a mixin done 🎉 👍

Glad to help expunge mixins 😄 They give me heartburn

**Pull request merged**